presto-spark-base/src/main/java/com/facebook/presto/spark/execution/property/NativeExecutionSystemConfig.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.execution.property;
+
+import com.facebook.airlift.configuration.Config;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+
+public class NativeExecutionSystemConfig
+{
+    private static final String CONCURRENT_LIFESPANS_PER_TASK = "concurrent-lifespans-per-task";
+    private static final String ENABLE_SERIALIZED_PAGE_CHECKSUM = "enable-serialized-page-checksum";
+    private static final String ENABLE_VELOX_EXPRESSION_LOGGING = "enable_velox_expression_logging";
+    private static final String ENABLE_VELOX_TASK_LOGGING = "enable_velox_task_logging";
+    private static final String HTTP_SERVER_HTTP_PORT = "http-server.http.port";
+    private static final String HTTP_EXEC_THREADS = "http_exec_threads";
+    private static final String NUM_IO_THREADS = "num-io-threads";
+    private static final String PRESTO_VERSION = "presto.version";
+    private static final String SHUTDOWN_ONSET_SEC = "shutdown-onset-sec";
+    private static final String SYSTEM_MEMORY_GB = "system-memory-gb";
+    private static final String TASK_MAX_DRIVERS_PER_TASK = "task.max-drivers-per-task";
+    private static final String DISCOVERY_URI = "discovery.uri";
```

Comment:
Also, do we have a cleaner way in C++ actually to have the config to have a source of truth so that both C++ and Java can refer to the name and values easily? Currently, we are duplicating code.

Reply:
Yeah we've tried to seek through some options and this way turned to be the cleanest.
The problem is we need to pass all the configs through cfgr. And current framework, in the main codepath it just takes those configs and populate them into config classes, then for the use of the code. Without config classes we cannot access these cfgr properties in "free form". Maybe we can but intrusive changes need to be made to the main codepath and framework.